tests/ui/vec.rs

```diff
@@ -176,3 +176,20 @@ fn below() {
         let _: String = a;
     }
 }
+
+fn func_needing_vec(_bar: usize, _baz: Vec<usize>) {}
+fn func_not_needing_vec(_bar: usize, _baz: usize) {}
+
+fn issue11861() {
+    macro_rules! this_macro_needs_vec {
+        ($x:expr) => {{ func_needing_vec($x.iter().sum(), $x) }};
```

Comment:
Can you also add a `for _ in $x {}` to this macro? This is to exercise the other case that this lints